src/states/common/base.rs

```diff
@@ -512,24 +512,18 @@ pub fn from_network_bytes(data: NetworkBytes) -> Result<Message, RoutingError> {
 
 /// A trait for types used to identify recipients of messages.
 pub trait MessageRecipient: Debug {
-    /// Resolve this recipient to a ConnectionInfo using the given PeerMap.
-    fn resolve<'a>(&'a self, peer_map: &'a PeerMap) -> Option<&'a ConnectionInfo>;
+    /// Resolve this recipient to a ConnectionInfo.
+    fn resolve(&self) -> Option<&ConnectionInfo>;
 }
 
-impl MessageRecipient for PublicId {
-    fn resolve<'a>(&'a self, peer_map: &'a PeerMap) -> Option<&'a ConnectionInfo> {
-        peer_map.get_connection_info(self)
-    }
-}
-
-impl MessageRecipient for XorName {
-    fn resolve<'a>(&'a self, peer_map: &'a PeerMap) -> Option<&'a ConnectionInfo> {
-        peer_map.get_connection_info(self)
+impl MessageRecipient for ConnectionInfo {
```

Comment:
I think we can probably remove this trait and go back to sending messages via `ConnectionInfo` only. Unless we want to keep the convenience of being able to send using both with `ConnectionInfo` and `P2pNode` which I'm not sure is worth the complexity.

Reply:
Good point. As these things are further simplified in Part 3 PR I'll see if it makes sense to just remote it in that PR

Reply:
Removed in https://github.com/maidsafe/routing/pull/1898